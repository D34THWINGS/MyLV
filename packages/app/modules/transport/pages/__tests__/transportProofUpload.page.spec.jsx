import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import { TransportProofPage } from '../transportProofUpload.page'

jest.unmock('../transportProofUpload.page')

describe('TransportProofPage', () => {
  let props

  const getWrapper = () => shallow(<TransportProofPage {...props} />)

  beforeEach(() => {
    props = {
      classes: {},
      valid: false,
      handleSubmit: jest.fn(),
      setExpirationDateToCurrentMonth: jest.fn(),
      proofExpirationDate: 0,
    }
  })

  it('should render properly', () => {
    // When
    const wrapper = getWrapper()

    // Then
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('should notify user when a valid proof is already uploaded', () => {
    // Given
    jest.spyOn(Date, 'now').mockImplementation(() => 42)
    props.proofExpirationDate = Date.now() + 69

    // When
    const wrapper = getWrapper()

    // Then
    expect(shallowToJson(wrapper)).toMatchSnapshot()
    Date.now.mockRestore()
  })
})
