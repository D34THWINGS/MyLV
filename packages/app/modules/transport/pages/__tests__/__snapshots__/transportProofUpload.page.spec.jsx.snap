// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransportProofPage should notify user when a valid proof is already uploaded 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Mon justificatif de transport | 
      MyLV
    </title>
  </HelmetWrapper>
  <WithStyles(Paper)>
    <WithStyles(List)>
      <WithStyles(ListItem)>
        <WithStyles(ListItemIcon)>
          <pure(InfoIcon) />
        </WithStyles(ListItemIcon)>
        <WithStyles(ListItemText)
          primary={
            <span>
              Tu as un justificatif de titre de transport valide jusqu'au 
              <b>
                01/01/1970
              </b>
            </span>
          }
        />
      </WithStyles(ListItem)>
    </WithStyles(List)>
  </WithStyles(Paper)>
  <WithStyles(Card)>
    <WithStyles(CardContent)>
      <WithStyles(Grid)
        container={true}
        spacing={8}
      >
        <WithStyles(Grid)
          item={true}
          xs={12}
        >
          <WithStyles(Typography)
            component="h2"
            gutterBottom={true}
            variant="h5"
          >
            Uploader un justificatif de titre de transport
          </WithStyles(Typography)>
          <WithStyles(Typography)
            gutterBottom={true}
          >
            Ce justificatif te permettra d'être remboursé pour ton titre de transport, merci donc de le fournir avant le 25 du mois. Renseigne aussi la date d'expiration du justificatif afin d'être rappellé automatiquement lorsque celui-ci arrive à expiration.
          </WithStyles(Typography)>
        </WithStyles(Grid)>
        <WithStyles(Grid)
          item={true}
          md={4}
          xs={12}
        >
          <Field
            autoFocus={true}
            component={[MockFunction]}
            fullWidth={true}
            label="Date de début de validité"
            name="startingDate"
            type="date"
          />
        </WithStyles(Grid)>
        <WithStyles(Grid)
          item={true}
          md={4}
          xs={12}
        >
          <Field
            component={[MockFunction]}
            fullWidth={true}
            label="Date d'expiration"
            name="expirationDate"
            type="date"
          />
        </WithStyles(Grid)>
        <WithStyles(Grid)
          item={true}
          md={4}
          xs={12}
        >
          <div>
            <WithStyles(Button)
              color="primary"
              onClick={[MockFunction]}
            >
              Valide pour ce mois
            </WithStyles(Button)>
          </div>
        </WithStyles(Grid)>
        <WithStyles(Grid)
          item={true}
          xs={12}
        >
          <Field
            accept="jpg,jpeg,JPG,JPEG,pdf,PDF,png,PNG"
            component={[MockFunction]}
            label="Choisir un fichier"
            name="file"
          />
        </WithStyles(Grid)>
      </WithStyles(Grid)>
    </WithStyles(CardContent)>
    <WithStyles(CardActions)>
      <WithStyles(Button)
        color="primary"
        disabled={true}
        size="small"
        type="submit"
      >
        Upload
      </WithStyles(Button)>
    </WithStyles(CardActions)>
  </WithStyles(Card)>
</form>
`;

exports[`TransportProofPage should render properly 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Mon justificatif de transport | 
      MyLV
    </title>
  </HelmetWrapper>
  <WithStyles(Card)>
    <WithStyles(CardContent)>
      <WithStyles(Grid)
        container={true}
        spacing={8}
      >
        <WithStyles(Grid)
          item={true}
          xs={12}
        >
          <WithStyles(Typography)
            component="h2"
            gutterBottom={true}
            variant="h5"
          >
            Uploader un justificatif de titre de transport
          </WithStyles(Typography)>
          <WithStyles(Typography)
            gutterBottom={true}
          >
            Ce justificatif te permettra d'être remboursé pour ton titre de transport, merci donc de le fournir avant le 25 du mois. Renseigne aussi la date d'expiration du justificatif afin d'être rappellé automatiquement lorsque celui-ci arrive à expiration.
          </WithStyles(Typography)>
        </WithStyles(Grid)>
        <WithStyles(Grid)
          item={true}
          md={4}
          xs={12}
        >
          <Field
            autoFocus={true}
            component={[MockFunction]}
            fullWidth={true}
            label="Date de début de validité"
            name="startingDate"
            type="date"
          />
        </WithStyles(Grid)>
        <WithStyles(Grid)
          item={true}
          md={4}
          xs={12}
        >
          <Field
            component={[MockFunction]}
            fullWidth={true}
            label="Date d'expiration"
            name="expirationDate"
            type="date"
          />
        </WithStyles(Grid)>
        <WithStyles(Grid)
          item={true}
          md={4}
          xs={12}
        >
          <div>
            <WithStyles(Button)
              color="primary"
              onClick={[MockFunction]}
            >
              Valide pour ce mois
            </WithStyles(Button)>
          </div>
        </WithStyles(Grid)>
        <WithStyles(Grid)
          item={true}
          xs={12}
        >
          <Field
            accept="jpg,jpeg,JPG,JPEG,pdf,PDF,png,PNG"
            component={[MockFunction]}
            label="Choisir un fichier"
            name="file"
          />
        </WithStyles(Grid)>
      </WithStyles(Grid)>
    </WithStyles(CardContent)>
    <WithStyles(CardActions)>
      <WithStyles(Button)
        color="primary"
        disabled={true}
        size="small"
        type="submit"
      >
        Upload
      </WithStyles(Button)>
    </WithStyles(CardActions)>
  </WithStyles(Card)>
</form>
`;
